examples/04_tool_inference_lock/ex_tool_inference_lock/main.rb

```diff
@@ -0,0 +1,199 @@
+# Copyright 2021 Trimble Inc
+# Licensed under the MIT license
+
+require 'sketchup.rb'
+require_relative 'vendor/inference_lock.rb'
```

Comment:
I also think we should stick with only `require` for the simplicity of these examples. Any new pattern we add will be a point where new onboarders might stop and wonder what the significance of the new pattern is.

I think we should stick to relative path from the root RB file.

Reply:
`require_relative` is ugly, but without it this tutorial can't be loaded without also adding the parent folder to the Ruby search path. For examples such script exists, but not tutorials.

I'm thinking about making these new additions example only, and not have them also being tutorials. It adds a lot of mental overhead to make sure the same changes are applied twice and that the modules and paths are all correct.

Reply:
This is resolved since I purged the "tutorial" and focused only on the "example".

Reply:
That sounds good. We can promote examples to tutorials any time later.

At some point we should sit down and talk about the tutorial series. Since I created the first onces I didn't have time to revisit. Originally I had ideas for a longer series that progressively explore various aspects of SketchUp and the API.